sonar/resources/projects/permissions.py

```diff
@@ -44,8 +45,19 @@ def query_filter(self, **kwargs):
         # TODO: Find a better way for handling typeahead calls..
         if current_user_record.is_moderator or (
                 request.args.get('q') and '.suggest' in request.args['q']):
-            return Q('term',
-                     metadata__organisation__pid=current_organisation['pid'])
+            must = [
+                Q('term',
+                  metadata__organisation__pid=current_organisation['pid'])
+            ]
+
+            # In suggestions only records published can be queried.
+            if request.args.get('q') and '.suggest' in request.args['q']:
+                must.append(
+                    Q('term',
+                      metadata__validation__status=Validation.STATUS_VALIDATED)
+                )
+
+            return Q('bool', must=must)
```

Comment:
using `.filter` should do the job.

Reply:
I don't know how to do that, because we seems not be able to manipulate the search object, but only build a query...

Reply:
`Q('bool', must=[term1, term2])` should be equivalent to `.filter('term', 'term1).filter('term', 'term2')`